### DISCUSSION TITLE
spaCy-LLM package integration with Azure OpenAI Service

### DISCUSSION THREAD

**Discussion by AashishTiwari** [explosion/spaCy#12653] [Help: Coding & Implementations] [answered]
Thanks for the latest LLM integration into spaCy. Need advice on whether the package https://github.com/explosion/spacy-llm will support Azure OpenAI Service in recent future or what changes are suggested (best practice) for a hack till there is official support.

One way to do this is by using LangChain https://github.com/hwchase17/langchain/blob/master/langchain/llms/openai.py and then use AzureOpenAI using Langchain? Just confirming if that's inline with best practice.

**Comment by rmitsch** [comment, marked as answer]
Hi @AashishTiwari! While we're planning to add native support for more APIs (next up are probably Cohere, PALM, Anthropic), I can't give you an exact date on when we'll do so for Azure's OpenAI API. 

So yes, our recommendation for using any API not natively supported is to use the `langchain` or `minichain` integration.
